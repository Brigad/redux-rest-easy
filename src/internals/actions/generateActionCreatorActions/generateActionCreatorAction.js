const generateActionCreatorAction = (cacheLifetime, actionType) => (
  normalizedURL,
  resourceId,
  normalizedPayload,
  metadata,
  principalResourceIds,
) => ({
  type: actionType,
  url: normalizedURL,
  resourceId,
  payload:
    normalizedPayload && Object.keys(normalizedPayload).length
      ? normalizedPayload
      : undefined,
  metadata,
  principalResourceIds:
    typeof principalResourceIds === 'string'
      ? [principalResourceIds]
      : principalResourceIds,
  cacheLifetime,
});

export default generateActionCreatorAction;
