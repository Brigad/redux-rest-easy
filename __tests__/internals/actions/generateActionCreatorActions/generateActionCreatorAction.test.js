import generateActionCreatorAction from '../../../../src/internals/actions/generateActionCreatorActions/generateActionCreatorAction';

const CACHE_LIFETIME = 0;
const ACTION_TYPE = '@@rest-easy/fruits/eat/REQUEST';
const NORMALIZED_URL = 'eat:https://api.co/fruits';
const ID = '2';
const ARRAY_ID = [ID];

describe('generateActionCreatorAction', () => {
  test('no args', () => {
    expect(generateActionCreatorAction()()).toMatchSnapshot();
  });

  test('no action type', () => {
    expect(generateActionCreatorAction(CACHE_LIFETIME)()).toMatchSnapshot();
  });

  test('no url, id, payload', () => {
    expect(
      generateActionCreatorAction(CACHE_LIFETIME, ACTION_TYPE)(),
    ).toMatchSnapshot();
  });

  test('no id, payload', () => {
    expect(
      generateActionCreatorAction(CACHE_LIFETIME, ACTION_TYPE)(NORMALIZED_URL),
    ).toMatchSnapshot();
  });

  test('no payload', () => {
    expect(
      generateActionCreatorAction(CACHE_LIFETIME, ACTION_TYPE)(
        NORMALIZED_URL,
        ID,
      ),
    ).toMatchSnapshot();
  });

  test('empty payload', () => {
    expect(
      generateActionCreatorAction(CACHE_LIFETIME, ACTION_TYPE)(
        NORMALIZED_URL,
        ID,
        {},
      ),
    ).toMatchSnapshot();
  });

  test('filled payload without principalResourceIds', () => {
    const payload = {
      fruits: {
        1: 'cherry',
      },
    };

    expect(
      generateActionCreatorAction(CACHE_LIFETIME, ACTION_TYPE)(
        NORMALIZED_URL,
        ID,
        payload,
      ),
    ).toMatchSnapshot();
  });

  test('filled payload, string principalResourceIds', () => {
    const payload = {
      fruits: {
        1: 'cherry',
      },
    };

    expect(
      generateActionCreatorAction(CACHE_LIFETIME, ACTION_TYPE)(
        NORMALIZED_URL,
        ID,
        payload,
        ID,
      ),
    ).toMatchSnapshot();
  });

  test('filled payload, array principalResourceIds', () => {
    const payload = {
      fruits: {
        1: 'cherry',
      },
    };

    expect(
      generateActionCreatorAction(CACHE_LIFETIME, ACTION_TYPE)(
        NORMALIZED_URL,
        ID,
        payload,
        ARRAY_ID,
      ),
    ).toMatchSnapshot();
  });
});
