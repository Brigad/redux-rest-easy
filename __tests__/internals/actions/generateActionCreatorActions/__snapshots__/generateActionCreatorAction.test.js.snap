// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateActionCreatorAction empty payload 1`] = `
Object {
  "cacheLifetime": 0,
  "metadata": undefined,
  "payload": undefined,
  "principalResourceIds": undefined,
  "resourceId": "2",
  "type": "@@rest-easy/fruits/eat/REQUEST",
  "url": "eat:https://api.co/fruits",
}
`;

exports[`generateActionCreatorAction filled payload without principalResourceIds 1`] = `
Object {
  "cacheLifetime": 0,
  "metadata": undefined,
  "payload": Object {
    "fruits": Object {
      "1": "cherry",
    },
  },
  "principalResourceIds": undefined,
  "resourceId": "2",
  "type": "@@rest-easy/fruits/eat/REQUEST",
  "url": "eat:https://api.co/fruits",
}
`;

exports[`generateActionCreatorAction filled payload, array principalResourceIds 1`] = `
Object {
  "cacheLifetime": 0,
  "metadata": Object {
    "resultsCount": 1,
  },
  "payload": Object {
    "fruits": Object {
      "1": "cherry",
    },
  },
  "principalResourceIds": Array [
    "2",
  ],
  "resourceId": "2",
  "type": "@@rest-easy/fruits/eat/REQUEST",
  "url": "eat:https://api.co/fruits",
}
`;

exports[`generateActionCreatorAction filled payload, string principalResourceIds 1`] = `
Object {
  "cacheLifetime": 0,
  "metadata": Object {
    "resultsCount": 1,
  },
  "payload": Object {
    "fruits": Object {
      "1": "cherry",
    },
  },
  "principalResourceIds": Array [
    "2",
  ],
  "resourceId": "2",
  "type": "@@rest-easy/fruits/eat/REQUEST",
  "url": "eat:https://api.co/fruits",
}
`;

exports[`generateActionCreatorAction no action type 1`] = `
Object {
  "cacheLifetime": 0,
  "metadata": undefined,
  "payload": undefined,
  "principalResourceIds": undefined,
  "resourceId": undefined,
  "type": undefined,
  "url": undefined,
}
`;

exports[`generateActionCreatorAction no args 1`] = `
Object {
  "cacheLifetime": undefined,
  "metadata": undefined,
  "payload": undefined,
  "principalResourceIds": undefined,
  "resourceId": undefined,
  "type": undefined,
  "url": undefined,
}
`;

exports[`generateActionCreatorAction no id, payload 1`] = `
Object {
  "cacheLifetime": 0,
  "metadata": undefined,
  "payload": undefined,
  "principalResourceIds": undefined,
  "resourceId": undefined,
  "type": "@@rest-easy/fruits/eat/REQUEST",
  "url": "eat:https://api.co/fruits",
}
`;

exports[`generateActionCreatorAction no payload 1`] = `
Object {
  "cacheLifetime": 0,
  "metadata": undefined,
  "payload": undefined,
  "principalResourceIds": undefined,
  "resourceId": "2",
  "type": "@@rest-easy/fruits/eat/REQUEST",
  "url": "eat:https://api.co/fruits",
}
`;

exports[`generateActionCreatorAction no url, id, payload 1`] = `
Object {
  "cacheLifetime": 0,
  "metadata": undefined,
  "payload": undefined,
  "principalResourceIds": undefined,
  "resourceId": undefined,
  "type": "@@rest-easy/fruits/eat/REQUEST",
  "url": undefined,
}
`;
